stories/__tests__/SQFormMultiSelect.stories.test.js

```diff
@@ -0,0 +1,189 @@
+    expect(options).toHaveLength(17); // initial '- -' + 16 MOCK_FRIENDS_OPTIONS
+  });
+
+  it('should render the multiselect with default initial value "- -"', async () => {
+    render(<SQFormMultiSelect size="auto" />);
+
+    const expandButton = await waitFor(() =>
+      screen.findByRole('button', {name: /friends/i})
+    );
+
+    expect(expandButton).toHaveTextContent(initialDropdownValue);
+  });
+
+  it('should render the multiselect with the correct options selected according to initialValues', async () => {
+    const initialValueLabels = 'Joe, Jane';
+
+    render(
+      <SQFormMultiSelect
+        size="auto"
+        SQFormProps={{
+          initialValues: {friends: [1, 2]}
+        }}
+      />
+    );
+
+    const expandButton = await waitFor(() =>
+      screen.findByRole('button', {name: /friends/i})
+    );
+
+    expect(expandButton).toHaveTextContent(initialValueLabels);
+  });
+
+  it('should show console warning if provided initial value not in options', async () => {
+    const consoleWarnSpy = jest
+      .spyOn(console, 'warn')
+      .mockImplementation(() => {});
+
+    render(
+      <SQFormMultiSelect
+        size="auto"
+        SQFormProps={{
+          initialValues: {friends: [0]}
+        }}
+      />
+    );
+
+    await waitFor(() =>
+      expect(consoleWarnSpy).toHaveBeenCalledWith(
+        expect.stringMatching(
+          /the requested display value.*could not be found/i
+        )
+      )
+    );
+
+    consoleWarnSpy.mockRestore();
+  });
+
+  it('should show empty list if no options passed', async () => {
+    const consoleWarnSpy = jest
+      .spyOn(console, 'warn')
+      .mockImplementation(() => {});
+
+    render(
+      <SQFormMultiSelect useSelectAll={false} size="auto" children={null} />
+    );
+
+    const expandButton = await screen.findByRole('button', {name: /friends/i});
+
+    userEvent.click(expandButton);
+    expect(screen.queryByRole('option')).not.toBeInTheDocument();
+    expect(consoleWarnSpy).toHaveBeenCalledWith(
+      expect.stringMatching(/the children you provided.*was undefined/i)
+    );
+
+    consoleWarnSpy.mockRestore();
+  });
+
+  it('should update when options are selected', async () => {
+    render(<SQFormMultiSelect size="auto" />);
+
+    const expandButton = await screen.findByRole('button', {name: /friends/i});
+
+    expect(expandButton).toHaveTextContent(initialDropdownValue);
+
+    userEvent.click(expandButton);
+
+    const option1 = screen.getByText('Jack');
+    userEvent.click(option1);
+
+    const option2 = screen.getByText('Jill');
+    userEvent.click(option2);
+
+    await waitFor(() => expect(expandButton).toHaveTextContent('Jack, Jill'));
+  });
+
+  it('should not open dropdown options if field is disabled', async () => {
+    render(<SQFormMultiSelect size="auto" isDisabled={true} />);
+
+    const expandButton = await screen.findByRole('button', {name: /friends/i});
+    expect(expandButton).toHaveAttribute('aria-disabled', 'true');
```

Comment:
I think the only thing that's missing is checking that the dropdown really won't open when clicked while disabled
```js
userEvent.click(expandButton);

const optionsList = screen.queryByRole('listbox');

expect(optionsList).not.toBeInTheDocument();
```

Reply:
> I think the only thing that's missing is checking that the dropdown really won't open when clicked while disabled
> 
> ```js
> userEvent.click(expandButton);
> 
> const optionsList = screen.queryByRole('listbox');
> 
> expect(optionsList).not.toBeInTheDocument();
> ```

awesome feedback. done!